From a0bf35c7c4be567589156e9c61c6fbfdfa5de1ec Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 24 Jun 2015 14:37:31 +0800
Subject: [PATCH] MLK-11163 dma: pxp-v2: refine the pxp scaling logic

The pxp scaling logic will cause issue when the s0
and out are both yuv formats. The issue has been
reported in another MM CR.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/dma/pxp/pxp_dma_v2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index 16ed2f2..5225bcf 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -657,6 +657,7 @@ static int pxp_set_scaling(struct pxps *pxp)
 	struct pxp_proc_data *proc_data = &pxp->pxp_conf_state.proc_data;
 	struct pxp_config_data *pxp_conf = &pxp->pxp_conf_state;
 	struct pxp_layer_param *s0_params = &pxp_conf->s0_param;
+	struct pxp_layer_param *out_params = &pxp_conf->out_param;
 
 	proc_data->scaling = 1;
 	decx = proc_data->srect.width / proc_data->drect.width;
@@ -676,6 +677,8 @@ static int pxp_set_scaling(struct pxps *pxp)
 			 (proc_data->drect.width * decx);
 	} else {
 		if (!is_yuv(s0_params->pixel_fmt) ||
+		    (is_yuv(s0_params->pixel_fmt) ==
+		     is_yuv(out_params->pixel_fmt)) ||
 		    (s0_params->pixel_fmt == PXP_PIX_FMT_GREY) ||
 		    (s0_params->pixel_fmt == PXP_PIX_FMT_GY04) ||
 		    (s0_params->pixel_fmt == PXP_PIX_FMT_VUY444)) {
-- 
1.8.0

